### DISCUSSION TITLE
Depth buffer with wgpu

### DISCUSSION THREAD

**Discussion by wrnrlr** [emilk/egui#2599] [Q&A] [answered]
I want to use a depth buffer on the wgpu backend. For this I've added a `DepthStencilState` to the `RenderPipelineDescriptor` of my pipeline. I should also be able to add a `RenderPassDepthStencilAttachment` to the render pass. However I can't find how to do this with the `egui_wgpu::CallbackFn`. The `paint` callback already has a render_pass configured but I need to add something like this:

```
depth_stencil_attachment: Some(wgpu::RenderPassDepthStencilAttachment {
        view: &self.depth_texture.view,
        depth_ops: Some(wgpu::Operations {
            load: wgpu::LoadOp::Clear(1.0),
            store: true,
        }),
        stencil_ops: None,
    }),
```

How can I do this?

**Comment by wrnrlr** [comment, marked as answer]
Got the answer, add this option:

```
eframe::NativeOptions {
         // ...
        depth_buffer: 1,
        #[cfg(feature = "wgpu")] renderer: eframe::Renderer::Wgpu,
        ..Default::default()
    };
```

**Comment by H4JUN** [comment]
@wrnrlr
I think I am also having the same issue. I am trying to add `RenderPassDepthStencilAttachment` to the render_pass, but the render_pass is passed as it is to the `paint` function in `CallbackTrait`.

Do you mind explaining more about how adding to the options that you mentioned as an answer allows you to attach DepthStencilAttachment to the render pass?

**Reply by LioQing** [reply]
I found it, the `depth_buffer` and `stencil_buffer` have to be in one the following supported values by wgpu:
https://github.com/emilk/egui/blob/e1f5d68995052442300b93ae0c6b85c3dea75255/crates/egui-wgpu/src/lib.rs#L372-L382

Also remember to set the `depth_stencil` field of `RenderPipelineDescriptor`.

**Reply by kongksora** [reply]
> I found it, the `depth_buffer` and `stencil_buffer` have to be in one the following supported values by wgpu:
> 
> https://github.com/emilk/egui/blob/e1f5d68995052442300b93ae0c6b85c3dea75255/crates/egui-wgpu/src/lib.rs#L372-L382
> 
> Also remember to set the `depth_stencil` field of `RenderPipelineDescriptor`.

Thanks! it helps a lot wen using wgpu for backend !!!!!
